src/zenml/cli/utils.py

```diff
@@ -1068,3 +1070,60 @@ def get_component_by_id_or_name_or_prefix(
                 f"No component of type `{component_type}` with name or id "
                 f"prefix '{id_or_name_or_prefix}' exists."
             )
+
+
+def get_execution_status_emoji(status: "ExecutionStatus") -> str:
+    """Returns an emoji representing the given execution status.
+
+    Args:
+        status: The execution status to get the emoji for.
+    
+    Returns:
+        An emoji representing the given execution status.
+    """
+    from zenml.enums import ExecutionStatus
+
+    if status == ExecutionStatus.FAILED:
+        return ":x:"
+    if status == ExecutionStatus.RUNNING:
+        return ":gear:"
+    if status == ExecutionStatus.COMPLETED:
+        return ":white_check_mark:"
+    if status == ExecutionStatus.CACHED:
+        return ":package:"
+    raise RuntimeError(f"Unknown status: {status}")
+
+
+def print_pipeline_runs_table(
+    client: Client, pipeline_runs: List["PipelineRunModel"]
+) -> None:
+    """Print a prettified list of all pipeline runs supplied to this method.
+
+    Args:
+        client: Repository instance
+        pipeline_runs: List of pipeline runs
+    """
+    runs_dicts = []
+    for pipeline_run in pipeline_runs:
+        try:
+            pipeline_name = client.zen_store.get_pipeline(
+                pipeline_run.pipeline_id
+            ).name
```

Comment:
This is potentially a lot of queries. Wondering if we could improve this somehow so that we're not iterating over dozens, potentially hundreds of these?

Reply:
For now it's fine, though, probably, so long as we don't overwhelm the server.

Reply:
If this really makes problems then we could rewrite it to query all pipelines, stacks, and users and then find the respective pipeline/user/stack for each pipeline run from the list. However, since that is quite convoluted, I would propose we keep it as is for now and only do this if it really causes issues.